docs/index-server.md

```diff
@@ -0,0 +1,128 @@
+the available executor list to find if any unassigned executor is
+present and use that executor for the current segment. If all the
+executors are assigned with some segment then find the least loaded
+executor on the basis of size.
+
+Initially the segment index size would be used to distribute the
+segments fairly among the executor because the actual cache size would
+be known to the driver only when the segments are cached and appropriate
+information is returned to the driver.
+
+**NOTE:** In case of legacy segment the index size if not available
+therefore all the legacy segments would be processed in a round robin
+fashion.
+
+After the job is completed the tasks would return the cache size held by
+each executor which would be updated to the executorToCacheMapping and
+the pruned blocklets which would be further used for result fetching.
+
+## Reallocation of executor
+In case executor(s) become dead/unavailable then the segments that were
+earlier being handled by those would be reassigned to some other
+executor using the distribution logic.
+
+**Note:** Cache loading would be done again in the new executor for the
+current query.
+
+## Fallback
+In case of any failure the index server would fallback to embedded mode
+which means that the JDBCServer would take care of distributed pruning.
+A similar job would be fired by the JDBCServer which would take care of
+pruning using its own executors. If for any reason the embedded mode
+also fails to prune the datamaps then the job would be passed on to
+driver.
+
+
+## Configurations
+
+##### carbon.properties 
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| carbon.enable.index.server       |  false | Enable the use of index server for pruning        |
+| carbon.index.server.ip |    NA   |   Specify the IP/HOST on which the server would be started. Better to specify the private IP. | 
+| carbon.index.server.port | NA | The port on which the index server has to be started. |
+| carbon.disable.index.server.fallback | false | Whether to enable/disable fallback for index server. Should be used for testing purposes only |
+|carbon.index.server.max.worker.threads| 500 | Number of RPC handlers to open for accepting the requests from JDBC driver.|
+|carbon.max.executor.lru.cache.size|  NA | Used to specify the max size for executor LRU cache.|
+|carbon.index.server.max.jobname.length|NA|The max length of the job to show in the index server application UI. For bigger queries this may impact performance as the whole string would be sent from JDBCServer to IndexServer.|
+
+##### spark-defaults.conf(only for secure mode)
+
+| Name     |      Default Value    |  Description |
+|:----------:|:-------------:|:------:       |
+| spark.carbon.indexserver.principal |  NA | Used for authentication, whether a valid application is trying to connect to the server or not. Set in both IndexServer and JDBCServer     |
```

Comment:
To be used in case of secure cluster only. It can also be mentioned that better for user to create a new principal that can be shared across users and would grant access only to IndexServer.

Reply:
done